src/instruction.rs

```diff
@@ -213,143 +213,239 @@ impl fmt::Display for MemoryReference {
-        r#type: GateType,
-    },
-    Declaration {
-        name: String,
-        size: Vector,
-        sharing: Option<String>,
-    },
-    Measurement {
-        qubit: Qubit,
-        target: Option<MemoryReference>,
-    },
-    Reset {
-        qubit: Option<Qubit>,
-    },
-    CalibrationDefinition(Calibration),
-    Capture {
-        blocking: bool,
-        frame: FrameIdentifier,
-        memory_reference: MemoryReference,
-        waveform: WaveformInvocation,
-    },
-    Delay {
-        duration: Expression,
-        frame_names: Vec<String>,
-        qubits: Vec<Qubit>,
-    },
-    Fence {
-        qubits: Vec<Qubit>,
-    },
-    FrameDefinition {
-        identifier: FrameIdentifier,
-        attributes: HashMap<String, AttributeValue>,
-    },
-    MeasureCalibrationDefinition {
-        qubit: Option<Qubit>,
-        parameter: String,
-        instructions: Vec<Instruction>,
-    },
-    Pragma {
-        name: String,
-        arguments: Vec<String>,
-        data: Option<String>,
-    },
-    Pulse {
-        blocking: bool,
-        frame: FrameIdentifier,
-        waveform: WaveformInvocation,
-    },
-    RawCapture {
-        blocking: bool,
-        frame: FrameIdentifier,
-        duration: Expression,
-        memory_reference: MemoryReference,
-    },
-    SetFrequency {
-        frame: FrameIdentifier,
-        frequency: f64,
-    },
-    SetPhase {
-        frame: FrameIdentifier,
-        phase: Expression,
-    },
-    SetScale {
-        frame: FrameIdentifier,
-        scale: Expression,
-    },
-    ShiftFrequency {
-        frame: FrameIdentifier,
-        frequency: f64,
-    },
-    ShiftPhase {
-        frame: FrameIdentifier,
-        phase: Expression,
-    },
-    SwapPhases {
-        frame_1: FrameIdentifier,
-        frame_2: FrameIdentifier,
-    },
-    WaveformDefinition {
-        name: String,
-        definition: Waveform,
-    },
-    Arithmetic {
-        operator: ArithmeticOperator,
-        destination: ArithmeticOperand,
-        source: ArithmeticOperand,
-    },
-    Halt,
-    Label(String),
-    Move {
-        destination: ArithmeticOperand,
-        source: ArithmeticOperand,
-    },
-    Exchange {
-        left: ArithmeticOperand,
-        right: ArithmeticOperand,
-    },
-    Load {
-        destination: MemoryReference,
-        source: String,
-        offset: MemoryReference,
-    },
-    Store {
-        destination: String,
-        offset: MemoryReference,
-        source: ArithmeticOperand,
-    },
-    Jump {
-        target: String,
-    },
-    JumpWhen {
-        target: String,
-        condition: MemoryReference,
-    },
-    JumpUnless {
-        target: String,
-        condition: MemoryReference,
-    },
+    Gate(Gate),
+    CircuitDefinition(CircuitDefinition),
+    GateDefinition(GateDefinition),
+    Declaration(Declaration),
+    Measurement(Measurement),
+    Reset(Reset),
+    CalibrationDefinition(CalibrationDefinition),
```

Comment:
If this one was already containing a single field, should we just leave it as is? Seems like an unnecessary extra layer.